src/app/pages/data-protection/replication/replication-form/replication-form.component.ts

```diff
@@ -107,18 +107,19 @@ export class ReplicationFormComponent implements OnInit {
     private replicationService: ReplicationService,
     private keychainCredentials: KeychainCredentialService,
     private authService: AuthService,
-    public slideInRef: SlideInRef<ReplicationTask | undefined, ReplicationTask | false>,
+    public slideInRef: SlideInOverlayRef,
+    @Inject(SLIDE_IN_DATA) protected existingReplication: ReplicationTask,
```

Comment:
I liked old approach better because it allowed us to have type safety for incoming and returning values.